skglm/datafits/base.py

```diff
@@ -31,6 +32,23 @@ def spec_to_float32(spec):
     return spec32
 
 
+@lru_cache
```

Comment:
This throws an error for python <= 3.7. 

```shell
TypeError: Expected maxsize to be an integer or None
```
It was corrected starting from 3.8

```suggestion
@lru_cache()
```